code/game/machinery/scav_selling_pad.dm

```diff
@@ -0,0 +1,205 @@
+/obj/machinery/scav_selling
+	name = "cargo hold pad"
+	icon = 'icons/obj/telescience.dmi'
+	icon_state = "lpad-idle-o"
+	var/idle_state = "lpad-idle-o"
+	var/warmup_state = "lpad-idle"
+	var/sending_state = "lpad-beam"
+	var/cargo_hold_id
+
+/obj/machinery/scav_selling/multitool_act(mob/living/user, obj/item/multitool/I)
+	if (istype(I))
+		to_chat(user, "<span class='notice'>You register [src] in [I]s buffer.</span>")
+		I.buffer = src
+		return TRUE
+
+/obj/machinery/computer/scav_selling_control
+	name = "cargo hold control terminal"
+	var/status_report = "Idle"
+	var/obj/machinery/scav_selling/pad
```

Comment:
This should use weakrefs.

Reply:
dont really know how to do those.
im mostly learning this as i work on it